### ISSUE TITLE
HTMX selection API inconsistent with Extended CSS Selectors

### ISSUE THREAD

**Issue by crisps-for-breakfast** [bigskysoftware/htmx#1816]
From the docs:

> Extended CSS Selectors
> `hx-target`, and most attributes that take a CSS selector, support an “extended” CSS syntax:
> 
> You can use the `this` keyword, which indicates that the element that the `hx-target` attribute is on is the target
> The `closest <CSS selector>` syntax will find the [closest](https://developer.mozilla.org/docs/Web/API/Element/closest) ancestor element or itself, that matches the given CSS selector. (e.g. `closest tr` will target the closest table row to the element)
> The `next <CSS selector>` syntax will find the next element in the DOM matching the given CSS selector.
> The `previous <CSS selector>` syntax will find the previous element in the DOM the given CSS selector.
> `find <CSS selector>` which will find the first child descendant element that matches the given CSS selector. (e.g find tr would target the first child descendant row to the element)
> In addition, a CSS selector may be wrapped in `<` and `/>` characters, mimicking the [query literal](https://hyperscript.org/expressions/query-reference/) syntax of hyperscript.
> 
> Relative targets like this can be useful for creating flexible user interfaces without peppering your DOM with loads of `id` attributes.

But the API only seems to provide the limited `htmx.find(target, selector)` and `htmx.closest(target, selector)`

Presumably there is some code somewhere that parses these Extended CSS Selectors and processes them, why isn't it available in the API? Perhaps as `htmx.select(target, selector)`?

**Comment by Telroshan** [comment]
You're right, `next` and `previous` aren't exposed, though an internal method exist (`next` resolves to [`scanForwardQuery`](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L605) and `previous` to [`scanBackwardsQuery`](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L615))

I can't think of any reason why we would **not** want to expose those in the API, so feel free to open a PR to add them!

**Comment by xhaggi** [comment]
We should think about whether it makes more sense to expose only one function that accept the extended CSS selector e.g. `htmx.select(target, extended css selector)`.

**Comment by 1cg** [comment]
no longer relevant
